### PR TITLE
Add a section on interactions of Pod priority and QoS

### PR DESCRIPTION
**Pull request by bsalamat**
Add a section to Pod Priority and Preemption page to explain the interactions of Pod priority and QoS classes.

### CODE REVIEW

File: content/en/docs/concepts/configuration/pod-priority-preemption.md [hunk 1]
```diff
@@ -368,4 +368,28 @@ When multiple nodes exist for preemption and none of the above scenarios apply,
...
+## Interactions of Pod priority and QoS
+
+Pod priority and
+[QoS](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/node/resource-qos.md)
+are two orthogonal features. They have little interactions and there is no
+default restrictions on settings priority of Pods based on their QoS classes.
+The scheduler's preemption logic does not take QoS into account when choosing
+preemption victims. Preemption considers Pod priority and tries to choose a set
+of victims with lowest priority. It considers other Pods for preemption only if
+the removal the lowest priority ones are not sufficient to schedule the
```
Comment (tengqm) [inline content/en/docs/concepts/configuration/pod-priority-preemption.md]: > the removal the lowest
the removal of the lowest

File: content/en/docs/concepts/configuration/pod-priority-preemption.md [hunk 2]
```diff
@@ -368,4 +368,28 @@ When multiple nodes exist for preemption and none of the above scenarios apply,
...
+preemptor Pod or the lowest priority ones are protected by PodDisruptionBudget.
+
+The only component that considers both QoS and Pod priority is
+[Kubelet out-of-resource eviction](https://kubernetes.io/docs/tasks/administer-cluster/out-of-resource/).
+"The kubelet ranks Pods for eviction first by whether or not their usage of the
+starved resource exceeds requests, then by Priority, and then by the consumption
+of the starved compute resource relative to the Pods’ scheduling requests."
+Please
+[see here](https://kubernetes.io/docs/tasks/administer-cluster/out-of-resource/#evicting-end-user-pods)
```
Comment (tengqm) [inline content/en/docs/concepts/configuration/pod-priority-preemption.md]: > https://kubernetes.io/docs/...
/docs/...

User relative paths please.

File: content/en/docs/concepts/configuration/pod-priority-preemption.md [hunk 3]
```diff
@@ -368,4 +368,28 @@ When multiple nodes exist for preemption and none of the above scenarios apply,
...
+are two orthogonal features. They have little interactions and there is no
+default restrictions on settings priority of Pods based on their QoS classes.
+The scheduler's preemption logic does not take QoS into account when choosing
+preemption victims. Preemption considers Pod priority and tries to choose a set
+of victims with lowest priority. It considers other Pods for preemption only if
+the removal the lowest priority ones are not sufficient to schedule the
+preemptor Pod or the lowest priority ones are protected by PodDisruptionBudget.
+
+The only component that considers both QoS and Pod priority is
+[Kubelet out-of-resource eviction](https://kubernetes.io/docs/tasks/administer-cluster/out-of-resource/).
```
Comment (tengqm) [inline content/en/docs/concepts/configuration/pod-priority-preemption.md]: > https://kubernetes.io/docs/
/docs

use relative paths please

File: content/en/docs/concepts/configuration/pod-priority-preemption.md [hunk 4]
```diff
@@ -368,4 +368,28 @@ When multiple nodes exist for preemption and none of the above scenarios apply,
...
+"The kubelet ranks Pods for eviction first by whether or not their usage of the
+starved resource exceeds requests, then by Priority, and then by the consumption
+of the starved compute resource relative to the Pods’ scheduling requests."
```
Comment (tengqm) [inline content/en/docs/concepts/configuration/pod-priority-preemption.md]: Why the quotation marks here?

File: content/en/docs/concepts/configuration/pod-priority-preemption.md [hunk 5]
```diff
@@ -368,4 +368,28 @@ When multiple nodes exist for preemption and none of the above scenarios apply,
...
+Please
+[see here](https://kubernetes.io/docs/tasks/administer-cluster/out-of-resource/#evicting-end-user-pods)
+for more details. This means that Kubelet does not evict Pods whose usage does
```
Comment (tengqm) [inline content/en/docs/concepts/configuration/pod-priority-preemption.md]: Remove "This means that"?

**Comment by mistyhacks** [comment]
/hold
(for feedback)

Please lift the hold when ready.

File: content/en/docs/concepts/configuration/pod-priority-preemption.md (hunk 5 above)
Comment (bsalamat) [reply content/en/docs/concepts/configuration/pod-priority-preemption.md]: done

**Comment by mistyhacks** [comment]
/assign

I'd like to add a copyedit commit, and I'll get to it tomorrow. Thanks!

**Comment by mistyhacks** [comment]
Added a copyedit. Lift the hold if it's good, or provide feedback if I changed any technical assertions with my edit. Thanks!

/approve
/lgtm
/hold

**Comment by bsalamat** [comment]
Looks good. Thanks @mistyhacks for your edits.

/hold cancel

**Pull request merged**